iothub/device/src/Transport/Mqtt/MqttLogger.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.Tracing;
+using MQTTnet.Diagnostics;
+
+namespace Microsoft.Azure.Devices.Client.Transport.Mqtt
+{
+    internal class MqttLogger : IMqttNetLogger
+    {
+        public bool IsEnabled => Logging.IsEnabled;
+
+        public void Publish(MqttNetLogLevel logLevel, string source, string message, object[] parameters, Exception exception)
```

Comment:
When you have some time, I'm interested to understand how you came to this implementation.